### PR TITLE
fix/connection tag again

### PR DESCRIPTION
**Pull request by blueMike16**

### CODE REVIEW

**Comment by haydenhuang** [comment]
Can one of the admins verify this patch?

File: model/Pool.js [hunk 1]
```diff
@@ -156,16 +181,19 @@ class Pool {
...
 	release() { }
 
-	_recycle(connection) {
-		const callback = this._connectionRequests.shift()
-		if (callback) {
-			Event.emit('recycle', connection)
-			connection.gotAt = new Date()
-			return callback(null, connection)
+	async _recycle(connection) {
+		for (const callback of this._connectionRequests) {
```
Comment (vivalalova) [inline model/Pool.js:185]: 只回收一個connection也要全看嗎

File: model/Pool.js [hunk 2]
```diff
@@ -156,16 +181,19 @@ class Pool {
...
+			if (Object.keys(this.connectionPool.using[callback.tag.name]).length < callback.tag.limit) {
+				Event.emit('recycle', connection)
+				this._connectionRequests.splice(this._connectionRequests.indexOf(callback), 1)
```
Comment (vivalalova) [inline model/Pool.js:188]: filter不好嗎?  一眼知道if在幹嘛 還不用for of

File: model/Pool.js [hunk 3]
```diff
@@ -36,7 +38,13 @@ class Pool {
 	}
 
 	get numberOfConnections() {
-		const amount = Object.keys(this.connectionPool.using).length + this.connectionPool.waiting.length
+		let count = 0
+
+		Object.keys(this.connectionPool.using).map(key => {
+			count = + Object.keys(this.connectionPool.using[key]).length
+		})
```
Comment (vivalalova) [inline model/Pool.js:45]: reduce

File: model/Pool.js [hunk 4]
```diff
@@ -87,47 +95,64 @@ class Pool {
...
 					return reject(err)
 				}
 				resolve(connection)
-			})
+			}, tag_name, limit)
 		})
 	}
 
-	getConnection(callback) {
+	getConnection(callback, tag_name = 'default', limit) {
```
Comment (vivalalova) [inline model/Pool.js:109]: 要不要都用object

**Pull request merged**